From 86534a23a31c223845ad4ee1d627c2ad38c52e93 Mon Sep 17 00:00:00 2001
From: Pi Delport <pjdelport@gmail.com>
Date: Sun, 13 Jul 2025 11:59:23 +0200
Subject: [PATCH] Support containers 0.8 (#317)

* Restrict Data.Tree imports

This avoids conflicts with Data.Tree.edges (added in containers 0.8).

* Increase containers upper bound to allow 0.8
---
 algebraic-graphs.cabal                      | 2 +-
 src/Algebra/Graph.hs                        | 2 +-
 src/Algebra/Graph/AdjacencyIntMap.hs        | 2 +-
 src/Algebra/Graph/AdjacencyMap.hs           | 2 +-
 src/Algebra/Graph/AdjacencyMap/Algorithm.hs | 2 +-
 src/Algebra/Graph/Class.hs                  | 2 +-
 src/Algebra/Graph/HigherKinded/Class.hs     | 2 +-
 src/Algebra/Graph/Relation.hs               | 2 +-
 src/Algebra/Graph/Relation/Symmetric.hs     | 2 +-
 9 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/algebraic-graphs.cabal b/algebraic-graphs.cabal
index d3d919b4..c9834199 100644
--- a/algebraic-graphs.cabal
+++ b/algebraic-graphs.cabal
@@ -69,7 +69,7 @@ source-repository head
 common common-settings
     build-depends:      array        >= 0.4     && < 0.6,
                         base         >= 4.11    && < 5,
-                        containers   >= 0.5.5.1 && < 0.8,
+                        containers   >= 0.5.5.1 && < 0.9,
                         deepseq      >= 1.3.0.1 && < 1.6,
                         transformers >= 0.4     && < 0.7
     default-language:   Haskell2010
diff --git a/src/Algebra/Graph.hs b/src/Algebra/Graph.hs
index 904701a3..d6a8d195 100644
--- a/src/Algebra/Graph.hs
+++ b/src/Algebra/Graph.hs
@@ -56,7 +56,7 @@ import Control.Monad.Trans.State (runState, get, put)
 import Data.Foldable (toList)
 import Data.Maybe (fromMaybe)
 import Data.String
-import Data.Tree
+import Data.Tree (Tree (..))
 import GHC.Generics
 
 import Algebra.Graph.Internal
diff --git a/src/Algebra/Graph/AdjacencyIntMap.hs b/src/Algebra/Graph/AdjacencyIntMap.hs
index f4a94280..e84b3c69 100644
--- a/src/Algebra/Graph/AdjacencyIntMap.hs
+++ b/src/Algebra/Graph/AdjacencyIntMap.hs
@@ -51,7 +51,7 @@ import Data.IntSet (IntSet)
 import Data.List ((\\))
 import Data.Monoid (Sum (..))
 import Data.Set (Set)
-import Data.Tree
+import Data.Tree (Forest, Tree (..))
 import GHC.Generics
 
 import qualified Data.IntMap.Strict as IntMap
diff --git a/src/Algebra/Graph/AdjacencyMap.hs b/src/Algebra/Graph/AdjacencyMap.hs
index 01fa0bf2..b075f307 100644
--- a/src/Algebra/Graph/AdjacencyMap.hs
+++ b/src/Algebra/Graph/AdjacencyMap.hs
@@ -55,7 +55,7 @@ import Data.Map.Strict (Map)
 import Data.Monoid
 import Data.Set (Set)
 import Data.String
-import Data.Tree
+import Data.Tree (Forest, Tree (..))
 import GHC.Generics
 
 import qualified Data.Map.Strict as Map
diff --git a/src/Algebra/Graph/AdjacencyMap/Algorithm.hs b/src/Algebra/Graph/AdjacencyMap/Algorithm.hs
index 00ceea46..8d9c0a97 100644
--- a/src/Algebra/Graph/AdjacencyMap/Algorithm.hs
+++ b/src/Algebra/Graph/AdjacencyMap/Algorithm.hs
@@ -33,7 +33,7 @@ import Data.Foldable (for_)
 import Data.Either
 import Data.List.NonEmpty (NonEmpty(..), (<|))
 import Data.Maybe
-import Data.Tree
+import Data.Tree (Forest, Tree (..), flatten, levels, unfoldForestM_BF)
 
 import Algebra.Graph.AdjacencyMap
 
diff --git a/src/Algebra/Graph/Class.hs b/src/Algebra/Graph/Class.hs
index 3c9dc968..fe1f72f7 100644
--- a/src/Algebra/Graph/Class.hs
+++ b/src/Algebra/Graph/Class.hs
@@ -46,7 +46,7 @@ module Algebra.Graph.Class (
     path, circuit, clique, biclique, star, tree, forest
     ) where
 
-import Data.Tree
+import Data.Tree (Forest, Tree (..))
 
 import Algebra.Graph.Label (Dioid, one)
 
diff --git a/src/Algebra/Graph/HigherKinded/Class.hs b/src/Algebra/Graph/HigherKinded/Class.hs
index 23c6fb9a..66185019 100644
--- a/src/Algebra/Graph/HigherKinded/Class.hs
+++ b/src/Algebra/Graph/HigherKinded/Class.hs
@@ -54,7 +54,7 @@ module Algebra.Graph.HigherKinded.Class (
 
 import Control.Applicative (Alternative(empty, (<|>)))
 import Control.Monad (MonadPlus, mfilter)
-import Data.Tree
+import Data.Tree (Forest, Tree (..))
 
 import qualified Algebra.Graph as G
 
diff --git a/src/Algebra/Graph/Relation.hs b/src/Algebra/Graph/Relation.hs
index bb02f2d3..e93864c9 100644
--- a/src/Algebra/Graph/Relation.hs
+++ b/src/Algebra/Graph/Relation.hs
@@ -46,7 +46,7 @@ import Control.DeepSeq
 import Data.Bifunctor
 import Data.Set (Set, union)
 import Data.String
-import Data.Tree
+import Data.Tree (Tree (..))
 import Data.Tuple
 
 import qualified Data.IntSet as IntSet
diff --git a/src/Algebra/Graph/Relation/Symmetric.hs b/src/Algebra/Graph/Relation/Symmetric.hs
index 184086cd..c33ea23e 100644
--- a/src/Algebra/Graph/Relation/Symmetric.hs
+++ b/src/Algebra/Graph/Relation/Symmetric.hs
@@ -45,7 +45,7 @@ import Control.DeepSeq
 import Data.Coerce
 import Data.Set (Set)
 import Data.String
-import Data.Tree
+import Data.Tree (Forest, Tree)
 
 import qualified Data.IntSet as IntSet
 import qualified Data.Set    as Set
